torchaudio/functional/functional.py

```diff
@@ -180,52 +188,49 @@ def griffinlim(
 
     specgram = specgram.pow(1 / power)
 
-    # randomly initialize the phase
-    batch, freq, frames = specgram.size()
+    # initialize the phase
     if rand_init:
-        angles = 2 * math.pi * torch.rand(batch, freq, frames)
+        angles = torch.rand(
+            specgram.size(),
+            dtype=_get_complex_dtype(specgram.dtype), device=specgram.device)
```

Comment:
wait the angles here should be a floating point tensor, no?

Reply:
No, `angles` is complex value. `Griffin-Lim` algorithm iteratively optimizes the phase (or the direction in complex plain) of each element of the given spectrogram so that at the end `istft` give the original waveform.

Reply:
*The original implementation was constructing the complex value Tensor called `angles` from real valued tensor called `angle` and `magnitude`.